### DISCUSSION TITLE
Deduplicate SSR preload links

### DISCUSSION THREAD

**Discussion by aaronadamsCA** [gatsbyjs/gatsby#32736] [Ideas / Feature Requests]
`<link as="fetch" rel="preload" .../>` entries in the `<head>` of SSR pages should be de-duplicated to reduce HTML payload size.

(There's a _chance_ this is causing iOS errors and/or log noise, which is how I noticed the duplication; I'm getting intermittent `Unexpected end of JSON input` errors in Sentry that seem to indicate JSON files are being parsed before they're downloaded. Wondering aloud if duplicate `preload` requests could trigger undefined Safari race condition behaviour.)

### Basic example

https://www.nishe.shop/ view source snippet shows two duplicate pairs of preload requests:

```html
<link as="fetch" rel="preload" href="/page-data/sq/d/1927784376.json" crossorigin="anonymous"/>
<link as="fetch" rel="preload" href="/page-data/sq/d/1927784376.json" crossorigin="anonymous"/>
<link as="fetch" rel="preload" href="/page-data/sq/d/4004863914.json" crossorigin="anonymous"/>
<link as="fetch" rel="preload" href="/page-data/sq/d/4004863914.json" crossorigin="anonymous"/>
```

I believe it's because multiple components on the page use the same static query hooks. Ideally this would be de-duplicated, since (a) reusable hooks are good and (b) smaller SSR pages are good.

**Comment by tcarter8** [comment]
Hi Aaron, I have the same issue with many duplicate preloads.  www.lucidmotors.com
I looked at nishe.shop and see that you no longer have duplicate json preloads, and am wondering if/how you fixed this?
Thanks in advance.
Tom

**Reply by aaronadamsCA** [reply]
Hey @tcarter8, I'm pretty sure it just disappeared through refactoring of our underlying components. Sorry I couldn't be more help!
